src/librustc_lexer/src/lib.rs

```diff
@@ -674,23 +696,51 @@ impl Cursor<'_> {
         has_digits
     }
 
-    fn float_exponent(&mut self) -> Result<(), ()> {
+    /// Eats the float exponent. Returns true if at least one digit was met,
+    /// and returns false otherwise.
+    fn eat_float_exponent(&mut self) -> bool {
         debug_assert!(self.prev() == 'e' || self.prev() == 'E');
-        if self.nth_char(0) == '-' || self.nth_char(0) == '+' {
+        if self.first() == '-' || self.first() == '+' {
             self.bump();
         }
-        if self.eat_decimal_digits() { Ok(()) } else { Err(()) }
+        self.eat_decimal_digits()
     }
 
-    // Eats the suffix if it's an identifier.
+    // Eats the suffix of the literal, e.g. "_u8".
     fn eat_literal_suffix(&mut self) {
-        if !is_id_start(self.nth_char(0)) {
+        self.eat_identifier();
+    }
+
+    // Eats the identifier.
+    fn eat_identifier(&mut self) {
+        if !is_id_start(self.first()) {
             return;
         }
         self.bump();
 
-        while is_id_continue(self.nth_char(0)) {
+        self.eat_identifier_continuation();
+    }
+
+    // Eats the identifier continuation.
+    // This method can be useful if the first symbol is allowed to be
+    // not an identifier start (e.g. number), but the rest of the token should
+    // contain only symbols valid for an identifier.
+    fn eat_identifier_continuation(&mut self) {
+        self.eat_while(is_id_continue);
```

Comment:
I'd inline this personally, `eat_while(is_id_continue)` is trivial and reads as good as `eat_identifier_continuation`.